frontend/src/tests/pages/MenuItemReview/MenuItemReviewPageCreate.test.js

```diff
@@ -0,0 +1,338 @@
+    const mockReview = {
+      itemId: 12,
+      item: { name: "Test Dish" },
+      itemsStars: 4,
+      // reviewerComments is omitted intentionally
+    };
+
+    axios.post.mockResolvedValueOnce({ data: mockReview });
+
+    renderPage({ idParam: "12" }); // ✅ use your real wrapper
+
+    fireEvent.click(screen.getByTestId("submit-button"));
+
+    await waitFor(() =>
+      expect(toast.success).toHaveBeenCalledWith(
+        `✅ Review submitted for "Test Dish"\n⭐ Rating: 4\n💬 Comment: No comments provided.`,
+        { autoClose: 8000 },
+      ),
+    );
+  });
+
+  it("shows 'not found' error on 404 + 'MenuItem'", async () => {
+    renderPage({ idParam: "88" });
+    axios.post.mockRejectedValue({
+      response: { status: 404, data: { message: "MenuItem missing" } },
+    });
+
+    fireEvent.click(screen.getByTestId("submit-button"));
+    await waitFor(() => expect(axios.post).toHaveBeenCalled());
+
+    expect(toast.error).toHaveBeenCalledWith("Menu item with ID 88 not found.");
+  });
+
+  it("falls back to generic error on 404 without 'MenuItem'", async () => {
+    renderPage({ idParam: "55" });
+    axios.post.mockRejectedValue({
+      response: { status: 404, data: { message: "Something else" } },
+    });
+
+    fireEvent.click(screen.getByTestId("submit-button"));
+    await waitFor(() => expect(axios.post).toHaveBeenCalled());
+
+    expect(toast.error).toHaveBeenCalledWith("Error creating review.");
+  });
+
+  it("falls back to generic error when no response object", async () => {
+    renderPage({ idParam: "99" });
+    axios.post.mockRejectedValue(new Error("Network failure"));
+
+    fireEvent.click(screen.getByTestId("submit-button"));
+    await waitFor(() => expect(axios.post).toHaveBeenCalled());
+
+    expect(toast.error).toHaveBeenCalledWith("Error creating review.");
+  });
+
+  it("does not crash when currentUser is undefined (optional-chaining guard)", () => {
+    renderPage({ currentUserData: undefined });
+    expect(() =>
+      fireEvent.click(screen.getByTestId("submit-button")),
+    ).not.toThrow();
+    // (We don’t re-check toast.error here, the other test covers that.)
+  });
+
+  //   it("falls back to 'No comments provided.' when response has no reviewerComments (kills non-optional-chain trim)", async () => {
+  //     // omit `reviewerComments` entirely
+  //     axios.post.mockResolvedValue({
+  //       data: {
+  //         item: { name: "Test" },
+  //         itemsStars: 1,
+  //         // no reviewerComments
+  //       },
+  //     });
+
+  //     fireEvent.click(screen.getByTestId("submit-button"));
+  //     await waitFor(() => expect(axios.post).toHaveBeenCalled());
+
+  //     expect(toast.success).toHaveBeenCalledWith(
+  //       `✅ Review submitted for "Test"\n⭐ Rating: 1\n💬 Comment: No comments provided.`,
+  //       { autoClose: 8000 },
+  //     );
+  //   });
```

Comment:
```suggestion

```
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed